include/ensmallen_bits/fw/proximal/proximal_impl.hpp

```diff
@@ -45,7 +45,7 @@ inline void Proximal::ProjectToL1Ball(MatType& v, double tau)
   MatType simplexSum = arma::cumsum(simplexSol);
 
   double nu = 0;
-  size_t rho = 0;
+  size_t rho = simplexSol.n_rows - 1;
```

Comment:
It looks also like the definition of `rho` could be pushed inside the loop.

Reply:
In this case we would have to calculate `theta` inside the loop as well, https://github.com/mlpack/ensmallen/blob/1cd90c6158d320f388b1ae6dd6cdfc6d15b632cb/include/ensmallen_bits/fw/proximal/proximal_impl.hpp#L56 maybe I'm missing something? I guess we could also ignore the warning because `rho` will always be `> 0` but I'I figured it might be easier to just initialize it once.

Reply:
You're right, how did I overlook that? 😄 Sorry about that.